framework/src/utils/RandomData.C

```diff
@@ -96,32 +96,29 @@ RandomData::updateGenerators()
   }
 
   if (_is_nodal)
-  {
-    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
-    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
+    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
```

Comment:
Because then you don't get consistent random numbers for different numbers
of processors. You need to generate the same sequence regardless of the
decomposition.
On Wed, Mar 16, 2016 at 9:50 AM Derek Gaston <notifications@github.com>
wrote:

> In framework/src/utils/RandomData.C
> <https://github.com/idaholab/moose/pull/6591#discussion_r56359020>:
>
> > @@ -96,32 +96,29 @@ RandomData::updateGenerators()
> >    }
> >
> >    if (_is_nodal)
> > -  {
> > -    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
> > -    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
> > +    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
>
> Why not just loop over local nodes and elements?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6591/files/a20165253410cfe9bfc05d3c3b04e1da92707c03#r56359020>
>